src/contract.rs

```diff
@@ -75,6 +78,36 @@ pub fn mint_path_msg(
     Ok(vec![wasm_msg1, wasm_msg2])
 }
 
+fn get_dens_owner(
+    querier: &QuerierWrapper,
+    token_id: String,
+    address_minting_the_path: String,
+) -> Option<String> {
+    let resp: Option<String> = querier
+        .query_wasm_smart(
+            &address_minting_the_path,
+            &Cw721QueryMsg::OwnerOf {
+                token_id: token_id,
+                include_expired: None,
+            },
+        )
+        .map(|resp: OwnerOfResponse| resp.owner)
+        .ok();
+
+    return resp;
+}
+
+fn is_in_claim_window(
+    path_root_claim_window: Option<u64>,
+    init_height: u64,
+    current_height: u64,
+) -> bool {
+    match path_root_claim_window {
+        Some(w) => init_height + u64::from(w) > current_height,
```

Comment:
tiny nit but use `checked_add`? Having said that I bet there's non checked add usage elsewhere in the code base >_<

Reply:
Is the current usage of `checked_add` fine, or would you use it differently? I mean the `if let Some ..`, or would you somehow unwrap it?

Reply:
I think that's an ok pattern :)